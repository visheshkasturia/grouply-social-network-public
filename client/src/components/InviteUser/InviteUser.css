.invite-status {
  text-align: center;
}