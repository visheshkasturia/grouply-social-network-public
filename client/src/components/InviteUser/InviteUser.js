/*eslint-disable */
import { useState } from "react";
import PostPosts from "../../apiCalls/PostPosts";
import { putInviteUser } from "../../apiCalls/putInviteUser";
import './InviteUser.css'

export default function InviteUser( {user, group} ) {
  const [inputs, setInputs] = useState({
    userToInvite: '',
  });
  const [status, setStatus] = useState();
  const handleChange = (event) => {
    const name = event.target.name;
    const value = event.target.value;
    setInputs(values => ({...values, [name]: value}))
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    putInviteUser(group._id, inputs.userToInvite).then((response) => {
      setStatus(response);
    });
    setInputs({
      userToInvite: '',
    });
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Enter Username to invite:
        <input 
          type="text" 
          name="userToInvite" 
          value={inputs.userToInvite} 
          onChange={handleChange}
        />
      </label>
      <br />
      <button type="submit"> Submit </button> 
      <div className="invite-status">{status}</div>
    </form>
  )
}
