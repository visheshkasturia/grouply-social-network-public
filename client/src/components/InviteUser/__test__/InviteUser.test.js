import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import InviteUser from '../InviteUser'

it('InviteUser renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<InviteUser  />, div);
});

it('InviteUser matches snapshot', () => {
  const tree = renderer.create(<InviteUser  />).toJSON();
  expect(tree).toMatchSnapshot();
});
