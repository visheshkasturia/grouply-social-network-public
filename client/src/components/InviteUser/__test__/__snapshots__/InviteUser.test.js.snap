// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteUser matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Enter Username to invite:
    <input
      name="userToInvite"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <button
    type="submit"
  >
     Submit 
  </button>
  <div
    className="invite-status"
  />
</form>
`;
